src/components/ReportsPage/AllScorecardsPage/AllScorecardsPage.tsx

```diff
@@ -13,31 +13,42 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import React, { useCallback } from 'react';
+import React, { ChangeEvent, useCallback, useState } from 'react';
 import { Content, ContentHeader } from '@backstage/core-components';
 import { Grid } from '@material-ui/core';
 import { AllScorecardsHeatmap } from './AllScorecardsHeatmap';
-import { useDropdown } from '../../../utils/hooks';
 import { GroupByOption } from '../../../api/types';
 import { GroupByDropdown } from '../Common/GroupByDropdown';
 import { CopyButton } from '../../Common/CopyButton';
-import { useLocation } from 'react-router-dom';
+import { useLocation, useNavigate, useSearchParams } from 'react-router-dom';
 import { buildUrl } from '../../../utils/URLUtils';
+import { stringifyUrl } from 'query-string';
 
 export const AllScorecardsPage = () => {
   const location = useLocation();
-  const queryParams = new URLSearchParams(useLocation().search);
+  const navigate = useNavigate();
+  const [searchParams] = useSearchParams();
 
-  const [groupBy, setGroupBy] = useDropdown<GroupByOption>(
-    (queryParams.get('groupBy') as GroupByOption) ?? GroupByOption.SERVICE,
+  const [groupBy, _setGroupBy] = useState<GroupByOption|undefined>(
```

Comment:
Why are we prefixing these setters with `_`? This is typically used for unused or private variables/methods (by convention), which doesn't seem to be the case here.

Reply:
Because it should not be used, except in the `setGoupBy`.

Reply:
So... it is used 😅